DependencyInjection/TrikoderOAuth2Extension.php

```diff
@@ -167,35 +174,36 @@ private function configureAuthorizationServer(ContainerBuilder $container, array
 
         if ($config['enable_client_credentials_grant']) {
             $authorizationServer->addMethodCall('enableGrantType', [
-                new Reference('league.oauth2.server.grant.client_credentials_grant'),
+                new Reference(ClientCredentialsGrant::class),
                 new Definition(DateInterval::class, [$config['access_token_ttl']]),
             ]);
         }
 
         if ($config['enable_password_grant']) {
             $authorizationServer->addMethodCall('enableGrantType', [
-                new Reference('league.oauth2.server.grant.password_grant'),
+                new Reference(PasswordGrant::class),
                 new Definition(DateInterval::class, [$config['access_token_ttl']]),
             ]);
         }
 
         if ($config['enable_refresh_token_grant']) {
             $authorizationServer->addMethodCall('enableGrantType', [
-                new Reference('league.oauth2.server.grant.refresh_token_grant'),
+                new Reference(RefreshTokenGrant::class),
                 new Definition(DateInterval::class, [$config['access_token_ttl']]),
             ]);
         }
 
         if ($config['enable_auth_code_grant']) {
             $authorizationServer->addMethodCall('enableGrantType', [
-                new Reference('league.oauth2.server.grant.auth_code_grant'),
+                new Reference(AuthCodeGrant::class),
                 new Definition(DateInterval::class, [$config['access_token_ttl']]),
             ]);
         }
 
         if ($config['enable_implicit_grant']) {
             $authorizationServer->addMethodCall('enableGrantType', [
-                new Reference('league.oauth2.server.grant.implicit_grant'),
+                new Reference('League\OAuth2\Server\Grant\ImplicitGrant'),
+                new Reference(ImplicitGrant::class),
```

Comment:
@rjwebdev Here you have now the ImplictGrant two times, on this line and on the one above.

Reply:
Sorry, again too fast. Removed first line